src/error_handling.jl

```diff
@@ -53,23 +53,37 @@ function handle_current_error(e, library, islast::Bool)
 end
 handle_current_error(e::NotInstalledError) = warn(string("lib ", e.library, " not installed, trying next library"))
 
+
 """
 Handles a list of thrown errors after no IO library was found working
 """
-function handle_error(exceptions::Vector)
+function handle_exceptions(exceptions::Vector, action)
+    # first show all errors when there are more then one
+    multiple = length(exceptions) > 1
+    println("Error$(multiple?"s" : "") encountered while $action.")
+    if multiple
+        println("All errors:")
+        for (err, file) in exceptions
+            println("   ", err)
+        end
+    end
+    # then handle all errors.
+    # this way first fatal exception throws and user can still see all errors
+    # TODO, don't throw, if it contains a NotInstalledError?!
+    println("Fatal error:")
     for exception in exceptions
-        continue_ = handle_error(exception...)
+        continue_ = handle_error(exception...) #
         continue_ || break
     end
 end
 
 handle_error(e, q) = rethrow(e)
 
 function handle_error(e::NotInstalledError, q)
-    println("Library ", e.library, " is not installed but can load format: ", q)
+    println("Library \"", e.library, "\" is not installed but is the recommended library to load format: \"", file_extension(q), "\"")
```

Comment:
*the* recommended library makes it sound like "most recommended" or "only recommended", but IIUC this could be printed for the 3rd library on the list of candidates. How about "recommended as a library to"?